test/unit/test_wait.py

```diff
@@ -0,0 +1,92 @@
+import logging
+
+import mocks
+import pytest
+from fake_statuses import MINIMAL_JSON, MINIMAL_STATUS
+
+import jubilant
+
+
+def test_ready_normal(run: mocks.Run, time: mocks.Time, caplog: pytest.LogCaptureFixture):
+    run.handle(['juju', 'status', '--format', 'json'], stdout=MINIMAL_JSON)
+    juju = jubilant.Juju()
+    caplog.set_level(logging.INFO, logger='jubilant')
+
+    status = juju.wait(lambda _: True)
+
+    assert run.call_count == 3
+    assert time.monotonic() == 2
+    assert status == MINIMAL_STATUS
+    assert len(caplog.records) == 1  # only logs on first call or when status changes
+    assert 'status changed' in caplog.text
+    assert 'mdl' in caplog.text
+
+
+def test_ready_glitch(run: mocks.Run, time: mocks.Time):
+    run.handle(['juju', 'status', '--format', 'json'], stdout=MINIMAL_JSON)
+    juju = jubilant.Juju()
+
+    n = 0
+
+    def ready_glitch(_: jubilant.Status):
+        nonlocal n
+        n += 1
+        return n != 2  # Glitch on second call
+
+    status = juju.wait(ready_glitch)
+
+    # Should wait for three successful calls to ready in a row:
+    # ready, not ready, ready, ready, ready (5 total)
+    assert run.call_count == 5
+    assert time.monotonic() == 4
+    assert status == MINIMAL_STATUS
+
+
+def test_modified_delay_and_successes(run: mocks.Run, time: mocks.Time):
+    run.handle(['juju', 'status', '--format', 'json'], stdout=MINIMAL_JSON)
+    juju = jubilant.Juju()
+
+    status = juju.wait(lambda _: True, delay=0.5, successes=5)
+
+    assert run.call_count == 5
+    assert time.monotonic() == 2.0
```

Comment:
Nit: I think it's a little confusing that this is the same as in the `ready_normal` test, except that it's a float. I get that it's because `(3-1)*1 == (5-1)*0.5`, but it took a minute to understand that. Maybe the delay could be 0.75 or something like that instead, so that the result is != 2?

Reply:
`2 == 2.0`, so I didn't actually need the `.0` there. But I've modified delay to 0.75 to avoid confusion.